### PR TITLE
Make Bundler/DuplicatedGem recognize different versions

### PR DESCRIPTION
**Pull request by iguchi1124**
Related issue: #3752

-----------------

Before submitting the PR make sure the following are checked:

* [ ] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [ ] All tests are passing.
* [ ] The new code doesn't generate RuboCop offenses.
* [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: spec/rubocop/cop/bundler/duplicated_gem_spec.rb [hunk 1]
```diff
@@ -85,5 +85,21 @@
...
+
+    context 'and same gems have different version specifications' do
+      let(:source) { <<-END }
+        if RUBY_VERSION < '2.2.5'
+          gem 'beaker', '~> 2.0', require: false
+          gem 'beaker-rspec', '~> 5.0', require: false
+        else
+          gem 'beaker-rspec', require: false
+        end
+      END
```
Comment (tejasbubane) [inline spec/rubocop/cop/bundler/duplicated_gem_spec.rb]: Does it register an offense if I don't have a conditional?
eg.
```
gem 'beaker-rspec'
gem 'beaker-rspec', '~> 5.0'
```

Also please check https://github.com/bbatsov/rubocop/issues/3752#issuecomment-267816609

Comment (iguchi1124) [inline spec/rubocop/cop/bundler/duplicated_gem_spec.rb]: @tejasbubane

I think it should returns error like ruby "syntax error".

But i don't have any good ideas at this point... 💭

**Comment by bbatsov** [comment]
In light of the additional comments on #3752 do we really need to support this?

**Comment by edestecd** [comment]
@bbatsov, yes unless you want the entire puppet module community to just turn off this check...

**Comment by bbatsov** [comment]
The other thing is that this PR simply doesn't solve the problem properly - it doesn't care if the gem versions are in separate version branches or not, which introduces some serious potential for issues.

**Comment by bbatsov** [comment]
Closing due to lack of progress.

**Pull request closed without merging**